pkg/platform/runtime/setup/setup.go

```diff
@@ -373,6 +414,22 @@ func (s *Setup) setupArtifact(buildEngine model.BuildEngine, a artifact.Artifact
 		return errs.Wrap(err, "Could not collect env info for artifact")
 	}
 
+	cnst := envdef.NewConstants(s.store.InstallPath())
```

Comment:
I thought we were doing this as one singular step at the end? ie. combine all the envdefs and then apply them.

Reply:
The applying of the file transforms can also be parallelized safely when it is just done inside the temporary installation directory.  So, I implemented it this way.

I also just realized that we are actually not merging the transforms. So, the global environment definition will not contain some of the transformations that we need to apply.

Reply:
The beauty of doing it all at the end is that it's automatically compatible with incremental updates. Is that still the case?

Reply:
It is, yes: Every artifact's transformations are going to be applied (whether pulled in incrementally or not), but inside their own sandbox, before the files are moved to the final destination.

Reply:
Another advantage (next to speed), is its predictability.  Theoretically, there could be two artifacts that install the same file, but with differing transformation directives.  When we do merge the transformations in the end, we might end up applying the wrong transformation directive, or too many or in the wrong order...
With this approach, we'll only apply the correct transformations.